dockerfile_parse/parser.py

```diff
@@ -341,12 +341,16 @@ def _instruction_getter(self, name, env_replace):
                 key_val_list = extract_labels_or_envs(env_replace=env_replace,
                                                       envs=envs,
                                                       instruction_value=instruction_desc['value'])
-                for key, value in key_val_list:
-                    if this_instruction == name:
-                        instructions[key] = value
-                        logger.debug("new %s %r=%r", name.lower(), key, value)
-                    if this_instruction == 'ENV':
-                        envs[key] = value
+                try:
+                    for key, value in key_val_list:
+                        if this_instruction == name:
+                            instructions[key] = value
+                            logger.debug("new %s %r=%r", name.lower(), key, value)
+                        if this_instruction == 'ENV':
+                            envs[key] = value
+                except KeyError:
```

Comment:
this just eats the exception and doesn't solve the root cause; I'm looking at this myself right now to see what's going on